app/controllers/vendor_api/decisions_controller.rb

```diff
@@ -149,5 +123,16 @@ def validate_course_is_in_current_recruitment_cycle!
         raise ValidationException, ["Course must be in #{RecruitmentCycle.current_year} recruitment cycle"]
       end
     end
+
+    def render_unauthorised_error(e)
+      render status: :unprocessable_entity, json: {
```

Comment:
This is a bit confusing. It was put in place to handle the make offer's unauthorised course error, and renders `The specified course is not associated with any of your organisations.` so vendors are able to understand why they were unable to make the offer.

I'm not sure though if a 422 should be returned for the other requests issued to the Decisions controller that result in `ProviderAuthorisation::NotAuthorisedError` as that is not how they are handled elsewhere e.g. deferrals.

Reply:
To address separately in https://trello.com/c/TLOEHD6g/4756-vendor-api-decisions-wrong-error-message